Unit tests

libuweber opened the pull request:
# Pull Request Template

## Description

Adds unit testing infrastructure and a few basic tests to start with.

## How has this been tested?
Unit Tests are integrated in the ci workflow.

**Versions**
* Microgateway: 2.0
* Helm Chart: 0.7.0

## Checklist:
- [ x] The code has been reviewed (self-review, ...).
- [ ] The parts of the code which are hard to understand are commented.
- [ ] The corresponding documentation has been updated.
- [x] The changes do not cause warnings.
- [ ] The spelling has been checked.

libuweber committed:
first test case

libuweber committed:
add test for metadata.labels

libuweber committed:
include unit test in ci workflow

libuweber committed:
fix unit test script

libuweber committed:
fix typo

libuweber committed:
fix job dependencies in ci.yaml

libuweber committed:
add checkout to workflow step

libuweber committed:
add checkout step 2

libuweber committed:
verify test failure breaks the build

libuweber committed:
fix broken test again

libuweber committed:
add unit test to checklist in the PR template

libuweber committed:
add basic unit tests for configmap using simple and expert DSL mode

libuweber committed:
developer notes added with information about unit tests.

libuweber committed:
remove todos

libuweber committed:
fixed version of chart-testing-action

and latest k8s versions

libuweber committed:
fix 1.18 k8s version

orltom commented on charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap:
```diff
@@ -0,0 +1,36 @@
+expert mode:
```
**subjective opinion**
I prefer when file name start with test_

libuweber replied:
I stuck to the naming conventions promoted by the unit testing tool we use. In this way, the tool is able to find the tests to run automatically. If we change the pattern, we would have to provide the test spec files to run in every invocation.

orltom reviewed (changes requested):
GitHub Actions fails

libuweber committed:
disable deployment in ci workflow

Microgateway 2.0 can not be deployed because the image is not yet available on dockerhub.

libuweber commented:
> GitHub Actions fails

deplyoment temporarily disabled in the ci workflow, as the microgateway 2.0 images are not yet available on docker hub.
